pilot/cmd/istioctl/config.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"istio.io/istio/pkg/log"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+var (
+
+	// TODO - Add support for non-kube istio deployments
+	// TODO - Add config-diff to get the difference between what pilot wants and what the proxy has
+	// TODO - Add support for non-default proxy config locations
+	configCmd = &cobra.Command{
+		Use:   "config <podID>",
+		Short: "Retrieves proxy configuration for the specified pod [kube only]",
+		Long: `
+Retrieves the proxy configuration for the specified pod when running in Kubernetes.
+Support for other environments to follow.
+`,
+		Example: `  # Retrieve config for productpage-v1-bb8d5cbc7-k7qbm pod
+  istioctl config productpage-v1-bb8d5cbc7-k7qbm`,
+		Aliases: []string{"conf"},
+		Args:    cobra.MinimumNArgs(1),
+		RunE: func(c *cobra.Command, args []string) error {
+			podName := args[0]
+			log.Infof("Retrieving proxy config for %q", podName)
+
+			ns := namespace
+			if ns == "" {
```

Comment:
consider replacing "" with v1.NamespaceAll

Reply:
In this case `""` indicates the namespace wasn't set and might be more appropriate.

Reply:
Whats the significance of v1.NamespaceAll?

Reply:
oh, that's the default of the variable 'namespace'. That's why I'm suggesting to use that instead of ""

Reply:
+1 to @baodongli .. Lets stick to the constants.